### DISCUSSION TITLE
paid VSIX licensing recommendations?

### DISCUSSION THREAD

**Discussion by mrlacey** [VsixCommunity/Community.VisualStudio.Toolkit#12] [General]
Have you built a paid for extension?
If so, how do you handle payment and licensing?
Any 3rd parties you'd recommend to help with this? (or recommend avoiding?)

Any other tips or advice as I consider my first paid (with trial) extensions?

**Comment by StevenRasmussen** [comment]
> Have you built a paid for extension?

Yes! [here](https://marketplace.visualstudio.com/items?itemName=SimplerSoftwareAdmin.XrmToolkit)

> If so, how do you handle payment?

The VSIX gallery is a good way to publish the extension and provide some basic info about it but I think ultimately if you're going to build a product you'll want a website to point people to for documentation as well as accepting payments (more on accepting payments in my answer regarding 3rd parties below).

> Licensing? 

This is always tricky.  I've found something that works pretty well for me. My product uses a license server (hosted on the website) to handle registering/unregistering a license key that is provided upon receiving payment.  I can also handle registering a license when the client does not have access to the internet... but I rarely need to use this method as most clients allow access to the internet.

>Any 3rd parties you'd recommend to help with this? (or recommend avoiding?)

The only 3rd parties I really use right now is with regards to accepting payment.  I have the option of accepting PayPal or a CC (which PayPal can do but for some reason, people were really confused and thought you HAD to have a PayPal account and so I provided this option).  I use [Authorize.Net](https://www.authorize.net/) for the CC integration and the API is decent but I haven't looked at options recently so there may be other better processors out there now.  I initially only started off with PayPal as they provide a pretty easy way to accept *most* forms of payment.

Good luck!  It's been worth it for me!

**Reply by mrlacey** [reply]
thanks.
